sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/ServiceTag.java

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.network.models;
+
+
+import com.azure.core.management.Region;
+import com.azure.core.util.CoreUtils;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Pattern;
+
+/**
+ * A service tag represents a group of IP address prefixes from a given Azure service.
+ * You can use service tags to define network access controls on network security groups, Azure Firewall, and user-defined routes.
+ * Use service tags in place of specific IP addresses when you create security rules and routes.
+ */
+public class ServiceTag {
+    private static final Map<String, ServiceTag> VALUES_BY_NAME = new ConcurrentHashMap<>();
+    /** Action Group. */
+    public static final ServiceTag ACTION_GROUP = new ServiceTag("ActionGroup", true, false, false, false);
+    /** Azure Storage.
+     * <p>Note: This tag represents the service, but not specific instances of the service.
+     * For example, the tag represents the Azure Storage service, but not a specific Azure Storage account.</p>
+     */
+    public static final ServiceTag STORAGE = new ServiceTag("Storage", false, true, true, true);
+    /**
+     * The IP address space that's outside the virtual network and reachable by the public internet.
+     * The address range includes the <a href="https://www.microsoft.com/en-us/download/details.aspx?id=41653">Azure-owned public IP address space</a>.
+     */
+    public static final ServiceTag INTERNET = new ServiceTag("Internet", true, true, false, false);
+    /**
+     * The virtual network address space (all IP address ranges defined for the virtual network),
+     * all connected on-premises address spaces, peered virtual networks, virtual networks connected to a virtual network gateway,
+     * the virtual IP address of the host, and address prefixes used on user-defined routes.
+     * This tag might also contain default routes.
+     */
+    public static final ServiceTag VIRTUAL_NETWORK = new ServiceTag("VirtualNetwork", true, false, false, false);
+
+    /**
+     * Get service tag from name.
+     *
+     * @param serviceTagName the name of the service tag, can be with or without region
+     * @return service tag, null if invalid
+     */
+    public static ServiceTag fromName(String serviceTagName) {
+        if (serviceTagName == null) {
+            return null;
+        }
+        String[] tagParts = serviceTagName.split(Pattern.quote("."));
+        if (tagParts.length == 1) { // root Tag
+            if (serviceTagName.contains(".")) {
+                return null;
+            }
```

Comment:
what can cause this?

Reply:
An invalid input `Storage.`. I added a test for this case [here](https://github.com/Azure/azure-sdk-for-java/pull/30804/commits/e67ea9c3912b70a151b558571a7bd52732a242b8#diff-5a6a8efaf8e19bd9973bdb22d1a2c7a2754b2acaf523882ad17e4970bd927b06R39).